crates/turborepo-lib/src/run/scope/target_selector.rs

```diff
@@ -187,34 +187,26 @@ pub enum InvalidSelectorError {
 pub fn is_selector_by_location(
     raw_selector: &str,
 ) -> Option<Result<AnchoredSystemPathBuf, InvalidSelectorError>> {
-    if !raw_selector.starts_with('.') {
-        return None;
-    }
+    let mut raw_selector_inner = raw_selector;
```

Comment:
I will not lie, this feels way too complicated for checking if `raw_selector` starts with `./`, `.\`, `../`, `..\` or if it is exactly `.`/`..`. (If it does something else, we need more comments) The Go code is a direct port of [JS](https://github.com/pnpm/pnpm/blob/fe322b67810dd6016bdc9ca636661a8e78710b4b/workspace/filter-workspace-packages/src/parsePackageSelector.ts#L63) and this feels like we're now different enough from JS that we should just embrace the difference and aim for readability. (Or go back to being a direct port of JS)


```
    let EXACT_MATCHES = [".", ".."];
    let PREFIXES = ["./", ".\\", "../", "..\\"];
    if EXACT_MATCHES.contains(&raw_selector)
        || PREFIXES
            .iter()
            .any(|prefix| raw_selector.starts_with(prefix))
    {
        Some(
            AnchoredSystemPathBuf::try_from(raw_selector)
                .map_err(|_| InvalidSelectorError::InvalidAnchoredPath(raw_selector.to_string())),
        )
    } else {
        None
    }
```

Could we get away with